File: problems/3-painted-wall/solution.cpp
#include <iostream>
#include <string>
using namespace std;

int brokenSequence(string wall)
{
    int count = 0;

    for (int i = 0; i < wall.length() - 1; i++)
    {
        string pair = wall.substr(i, 2);

        if (pair == "RG" || pair == "GB" || pair == "BR")
        {
            continue;
        }
        else
        {
            count++;
        }
    }

    return count;
}

int main()
{
    string text;
    getline(cin, text);
    cout << brokenSequence(text) << endl;
    return 0;
}

File: problems/3-painted-wall/solution.py
def broken_sequence(wall: str) -> int:
    count = 0

    for i in range(len(wall) - 1):
        pair = wall[i : i + 2]

        if pair == "RG" or pair == "GB" or pair == "BR":
            continue
        else:
            count += 1

    return count


text = input()
print(broken_sequence(text))
